### ISSUE TITLE
Comment at the end of input causes an exception when `parseNext()`

### ISSUE THREAD

**Issue by HyukjinKwon** [uniVocity/univocity-parsers#213]
Here is the reproducer:

```java
package com.mobigen.hadooptest;

import java.io.StringReader;

import com.univocity.parsers.csv.CsvParser;
import com.univocity.parsers.csv.CsvParserSettings;

class Main {
    public static void main(String[] args) throws Exception {
        CsvParserSettings settings = new CsvParserSettings();
        settings.getFormat().setLineSeparator("\n");
        settings.getFormat().setComment('#');
        CsvParser parser = new CsvParser(settings);

        // OK
        parser.beginParsing(new StringReader("#comment\na\nb"));
        System.out.println(parser.parseNext()[0]); // a
        System.out.println(parser.parseNext()[0]); // b
        System.out.println(parser.parseNext()); // null

        // Failed
        parser.beginParsing(new StringReader("a\nb\n#comment"));
        System.out.println(parser.parseNext()[0]); // a
        System.out.println(parser.parseNext()[0]); // b 
        System.out.println(parser.parseNext()); // Exception
    }
}
```

```
a
b
null
a
b
Exception in thread "main" com.univocity.parsers.common.TextParsingException: java.lang.IllegalArgumentException - Unable to skip 1 lines from line 2. End of input reached
Parser Configuration: CsvParserSettings:
	Auto configuration enabled=true
	Autodetect column delimiter=false
	Autodetect quotes=false
	Column reordering enabled=true
	Empty value=null
	Escape unquoted values=false
	Header extraction enabled=null
	Headers=null
	Ignore leading whitespaces=true
	Ignore trailing whitespaces=true
	Input buffer size=1048576
	Input reading on separate thread=true
	Keep escape sequences=false
	Keep quotes=false
	Length of content displayed on error=-1
	Line separator detection enabled=false
	Maximum number of characters per column=4096
	Maximum number of columns=512
	Normalize escaped line separators=true
	Null value=null
	Number of records to read=all
	Processor=none
	Restricting data in exceptions=false
	RowProcessor error handler=null
	Selected fields=none
	Skip bits as whitespace=true
	Skip empty lines=true
	Unescaped quote handling=nullFormat configuration:
	CsvFormat:
		Comment character=#
		Field delimiter=,
		Line separator (normalized)=\n
		Line separator sequence=\n
		Quote character="
		Quote escape character="
		Quote escape escape character=null
Internal state when error was thrown: line=3, column=0, record=2, charIndex=12, headers=[a]
	at com.univocity.parsers.common.AbstractParser.handleException(AbstractParser.java:364)
	at com.univocity.parsers.common.AbstractParser.parseNext(AbstractParser.java:508)
	at com.mobigen.hadooptest.Main.main(Main.java:25)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)
Caused by: java.lang.IllegalArgumentException: Unable to skip 1 lines from line 2. End of input reached
	at com.univocity.parsers.common.input.AbstractCharInputReader.skipLines(AbstractCharInputReader.java:295)
	at com.univocity.parsers.common.AbstractParser.processComment(AbstractParser.java:102)
	at com.univocity.parsers.common.AbstractParser.parseNext(AbstractParser.java:470)
	... 6 more
```

I expected the last 

```
...
System.out.println(parser.parseNext()); // Exception
...
```

prints `null` rather than throwing an exception but it looks causing an error. In my case (in Spark's case, as you might already know), it checks `null` to stop parsing further, just like - https://github.com/uniVocity/univocity-parsers#to-read-all-rows-of-a-csv-iterator-style

Seems a small bug to me. Could you confirm this please?

**Comment by jbax** [comment]
Interesting, I'm getting the correct behaviour here. This was fixed yesterday in this commit: https://github.com/uniVocity/univocity-parsers/commit/cd066141eff78cab5b4c0796fd9538ba41caaa7f 

That commit is part of the 2.5.9 release.  Do you mind double-checking that you are actually running your test with version 2.5.9? Depending on your environment you might have an older version being referenced by a stale dependency cache or something.

**Comment by HyukjinKwon** [comment]
Oh, maybe my mistake. Let me double check when I get my home.

**Comment by HyukjinKwon** [comment]
Yea, indeed this is fixed in 2.5.9. Sorry, it was my mistake. Thanks for confirming this.
